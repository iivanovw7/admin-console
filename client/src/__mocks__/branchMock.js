export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};