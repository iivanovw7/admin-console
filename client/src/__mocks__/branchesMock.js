export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
