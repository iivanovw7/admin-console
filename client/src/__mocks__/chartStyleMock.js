export const style = {
  type: "PieChart",
  id: 2
};
