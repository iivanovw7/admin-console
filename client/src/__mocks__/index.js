export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';
export * from './groupMock';
export * from './groupsMock';
export * from './roleMock';
export * from './rolesMock';
export * from './ticketMock';
export * from './ticketsMock';
export * from './messageMocks';
export * from './messagesMocks';
export * from './statisticsMock';
