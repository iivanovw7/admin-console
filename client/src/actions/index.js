export * from './authActions';
export * from './branchesActions';
export * from './groupsActions';
export * from './rolesActions';
export * from './usersActions';
export * from './ticketsActions';
export * from './messagesActions';
export * from './statisticsActions';