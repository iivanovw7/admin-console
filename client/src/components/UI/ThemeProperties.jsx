import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

