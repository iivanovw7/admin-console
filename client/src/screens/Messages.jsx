import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { CssBaseline, Typography } from '@material-ui/core';

const Messages = props => {

  const { classes, history } = props;

  return (
    <main className={classes.content}>
      <div className={classes.toolbar}/>
      <h2>Messages</h2>
      <Typography paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
        incididunt ut labore et dolore magna aliqua. Rhoncus dolor purus non enim praesent
        elementum facilisis leo vel. Risus at ultrices mi tempus imperdiet. Semper risus in
        hendrerit gravida rutrum quisque non tellus. Convallis convallis tellus id interdum
        velit laoreet id donec ultrices. Odio morbi quis commodo odio aenean sed adipiscing.
        Amet nisl suscipit adipiscing bibendum est ultricies integer quis. Cursus euismod quis
        viverra nibh cras. Metus vulputate eu scelerisque felis imperdiet proin fermentum leo.
      </Typography>
      <Typography paragraph>
        Consequat mauris nunc congue nisi vitae suscipit. Fringilla est ullamcorper eget nulla
        facilisi etiam dignissim diam. Pulvinar elementum integer enim neque volutpat ac
        tincidunt. Ornare suspendisse sed nisi lacus sed viverra tellus. Purus sit amet volutpat
        consequat mauris. Elementum eu facilisis sed odio morbi. Euismod lacinia at quis risus
        sed vulputate odio. Morbi tincidunt ornare massa eget egestas purus viverra accumsan in.
        In hendrerit gravida rutrum quisque non tellus orci ac. Pellentesque nec nam aliquam sem
        et tortor. Habitant morbi tristique senectus et. Adipiscing elit duis tristique
      </Typography>
    </main>
  );
};

Messages.propTypes = {
  classes: PropTypes.object.isRequired
};

export default Messages;



