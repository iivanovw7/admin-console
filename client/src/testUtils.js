import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};