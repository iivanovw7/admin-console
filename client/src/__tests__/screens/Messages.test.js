import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import { createMemoryHistory } from 'history';
import jest from 'jest-mock';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import { getMessages } from '../../actions';
import { Wrapper } from '../../components/UI/ThemeProperties';
import Messages from '../../screens/Messages';
import { storeFactory } from '../../testUtils';
import * as mocks from './../../__mocks__/';

const history = createMemoryHistory('/dashboard');

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);

  const Composition = connect(null, { getMessages })(withStyles(Wrapper, { withTheme: true })(withRouter(Messages)));
  const props = {
    history: history,
    messages: {
      list: {
        output: mocks.MESSAGES
      }
    }
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders messages main screen form with styles,
   *  Checks if component rendered properly.
   */
  describe('Messages screen', () => {
    it('Should render HOC with styles properly', () => {
      const mockFunc = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Composition {...props} dispatch={mockFunc}/>
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.html()).not.toBe(null);
      expect(wrapper.html()).not.toBe(undefined);
      expect(wrapper.length).toEqual(1);
      expect(wrapper.find('h2').text()).toBe('Messages');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();



