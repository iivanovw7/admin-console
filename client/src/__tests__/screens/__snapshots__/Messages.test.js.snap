// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages screen Should render HOC with styles properly 1`] = `"<main class=\\"withRouter-Messages--contentList-19 withRouter-Connect-WithStyles-withRouter-Messages-----contentList-5\\"><div><div class=\\"MuiPaper-root-29 MuiPaper-elevation2-33 MuiPaper-rounded-30 withRouter-Messages--titleContainer-24 withRouter-Connect-WithStyles-withRouter-Messages-----titleContainer-10\\"><div class=\\"withRouter-Messages--selectorsContainer-26 withRouter-Connect-WithStyles-withRouter-Messages-----selectorsContainer-12\\"><h2>Messages</h2></div><div class=\\"ButtonAddNew-formRoot-56\\"><button class=\\"MuiButtonBase-root-85 MuiButton-root-59 MuiButton-text-61 MuiButton-flat-64 MuiButton-sizeSmall-82 ButtonAddNew-mobile-57\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-60\\"><strong>Create</strong></span><span class=\\"MuiTouchRipple-root-268\\"></span></button><button class=\\"MuiButtonBase-root-85 MuiFab-root-88 MuiFab-primary-90 MuiFab-sizeSmall-96 ButtonAddNew-desktop-58\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Add\\"><span class=\\"MuiFab-label-89\\"><svg class=\\"MuiSvgIcon-root-98\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-268\\"></span></button></div></div><div class=\\"MuiPaper-root-29 MuiPaper-elevation1-32 MuiPaper-rounded-30 SearchBar-searchRoot-116\\"><div style=\\"color: rgba(0, 0, 0, 0.54);\\"><span class=\\"material-icons MuiIcon-root-129\\" aria-hidden=\\"true\\" title=\\"Search by User\`s Name, Surname or Subject\\">help_outline</span></div><hr class=\\"MuiDivider-root-138 SearchBar-divider-117\\"><div class=\\"MuiInputBase-root-143 SearchBar-inputRoot-119\\"><input class=\\"MuiInputBase-input-153 MuiInputBase-inputType-156 MuiInputBase-inputTypeSearch-157\\" placeholder=\\"Searchâ€¦\\" type=\\"search\\" value=\\"\\"></div><svg class=\\"MuiSvgIcon-root-98\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\" style=\\"color: rgba(0, 0, 0, 0.54);\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></div><div class=\\"Spinner-spinnerBlock-161\\"><div class=\\"MuiCircularProgress-root-174 MuiCircularProgress-colorPrimary-177 MuiCircularProgress-indeterminate-176\\" style=\\"width: 40px; height: 40px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-179\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-180 MuiCircularProgress-circleIndeterminate-182\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div></div><div class=\\"MuiPaper-root-29 MuiPaper-elevation2-33 MuiPaper-rounded-30 withRouter-Messages--controlsContainer-25 withRouter-Connect-WithStyles-withRouter-Messages-----controlsContainer-11\\" style=\\"margin-top: 24px; margin-bottom: 24px;\\"><div class=\\"withRouter-Messages--formLimitSelector-22 withRouter-Connect-WithStyles-withRouter-Messages-----formLimitSelector-8\\"><div class=\\"MuiFormControl-root-184 withRouter-Messages--formControl-23 withRouter-Connect-WithStyles-withRouter-Messages-----formControl-9\\"><label class=\\"MuiFormLabel-root-199 MuiFormLabel-filled-203 MuiInputLabel-root-188 MuiInputLabel-formControl-193 MuiInputLabel-animated-196 MuiInputLabel-shrink-195\\" data-shrink=\\"true\\">Results limit</label><div class=\\"MuiInputBase-root-143 MuiInput-root-213 MuiInput-underline-217 MuiInputBase-formControl-144 MuiInput-formControl-214\\"><div class=\\"MuiSelect-root-206\\"><select class=\\"MuiSelect-select-207 MuiInputBase-input-153 MuiInput-input-221\\" aria-invalid=\\"false\\"><option value=\\"10\\">10</option><option value=\\"20\\">20</option><option value=\\"30\\">30</option><option value=\\"50\\">50</option><option value=\\"60\\">60</option></select><svg class=\\"MuiSvgIcon-root-98 MuiSelect-icon-212\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div></div></div><div class=\\"withRouter-Messages--formRoot-20 withRouter-Connect-WithStyles-withRouter-Messages-----formRoot-6\\"><div style=\\"display: flex; margin: 5px; align-items: center;\\"><div class=\\"MuiCircularProgress-root-174 MuiCircularProgress-colorPrimary-177 MuiCircularProgress-indeterminate-176\\" style=\\"width: 24px; height: 24px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-179\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-180 MuiCircularProgress-circleIndeterminate-182\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div><div style=\\"display: flex; align-items: center; margin: 5px;\\"><button class=\\"MuiButtonBase-root-85 MuiButtonBase-disabled-86 MuiIconButton-root-226 MuiIconButton-colorPrimary-228 MuiIconButton-disabled-230\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Back\\"><span class=\\"MuiIconButton-label-231\\"><svg class=\\"MuiSvgIcon-root-98\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span></button></div><p class=\\"MuiTypography-root-232 MuiTypography-body1-241 withRouter-Messages--pageSelectorTitle-21 withRouter-Connect-WithStyles-withRouter-Messages-----pageSelectorTitle-7\\"><strong>1 of 1</strong></p><div style=\\"display: flex; align-items: center; margin: 5px;\\"><button class=\\"MuiButtonBase-root-85 MuiButtonBase-disabled-86 MuiIconButton-root-226 MuiIconButton-colorPrimary-228 MuiIconButton-disabled-230\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Next\\"><span class=\\"MuiIconButton-label-231\\"><svg class=\\"MuiSvgIcon-root-98\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span></button></div></div></div></main>"`;
